enhancements/etcd/disaster-recovery-with-ceo.md

```diff
@@ -19,86 +19,130 @@ see-also:
+See the implementation details below.
+    
+### Recovering from a loss of quorum
+Recovering from a loss of quorum requires restoring from a previous backup. See the implementation details for `Restoring the cluster from backup`.
+   
+### Recovery of a member with a bad data-dir.
+
+###  All nodes being shut off at the same time and restarted.
+Fix the initialization process to detect lights-out scenario to restart the etcd without requiring any manual intervention.
+    
+### Upgrade, downgrade, re-upgrade
+Upgrade to 4.4, downgrade to 4.3, and reupgrading to 4.4 again should be supported.
 
 ## Implementation Details/Notes/Constraints
 
-*  Fix the version mismatch bug -- **done**.
-*  Change the name of the spec from `etcd-member.yaml` to `etcd-pod.yaml` -- **done**.
-*  Change the backup scripts to include the static pod resources of etcd -- **done**.
-*  Change the restore scripts to extract the static pod resources of etcd -- **done**.
-*  Fix the need to extract ETCD_DNS_NAME and  DISCOVERY_DOMAIN from pod spec -- **done**.
+### Restoring the cluster from backup
+In the prior releases, we restored the backup on all masters simultaneously for the cluster to come backup properly. With this release we implement a different approach of bringing a control plane up with a single node etcd cluster and then scaling it up to three.
+
+This new approach involves two steps:
+1. Get the first etcd member running. 
+
+    1. move the existing etcd pod yaml out of kubelet manifest directory and move etcd data directory into a different 
+  location on all the master nodes. This will stop all the etcds leading to total quorum loss and remove all of the data
+   directories, making the backup a single source of truth.
+    1. Select a master node to run the restore operation on (recovery node) and make sure etcd backup file in present
+   on the recovery node
+    1. copy the snapshot file on a temporary location and move the copy to backup directory to 
+  /var/lib/etcd-backup/snapshot.db on the recovery node
+    1. move the kube-apiserver backed up revision back into kubelet manifest directory and delete other revisions and 
+  related resources(certs and configmaps). Because the restoration is happening from a backed up data, the revision of
+  kube-apiserver pod yaml needs to match the revision that the operator was on while taking the backup.
+    1. copy the restore-etcd-pod.yaml from backed up resources to /etc/kubernetes/manifests/etcd-pod.yaml. This will 
+  restore a single member control plane using the backed up snapshot.db and then start the etcd pod bringing the control
+  plane up. 
+
+2. Get the operator to deploy static pods that can decide that they are being restored, so they don't start on the existing data-dir.
+    1. force CEO redeployment via `oc patch forceredpeloymment=reason-time`
+    1. When CEO redeployment is forced, the existing nodes will start with new pods, a scale up similar to bootstrap will take place
```

Comment:
new revision does not scale etcd only restart the static pods in rolling fasion with new spec.